web3.go

```diff
@@ -334,10 +345,34 @@ func ConvertArguments(method abi.Method, inputParams []interface{}) []interface{
 			switch inputParams[i].(type) {
 			case *big.Int:
 				val = inputParams[i].(*big.Int)
+				convertedParams = append(convertedParams, val)
+
+			case float64: // convenient for taking args directly from JSON values
```

Comment:
Would using `json.Number` make this simpler?

Reply:
JSON values with `interface{}` fields are parsed into float64.

Reply:
That default behavior can be changed with `Decoder.UseNumber` https://golang.org/pkg/encoding/json/#Decoder.UseNumber.

Reply:
I think the problem with that is you'd have to tell everyone to do that before calling this function. I suppose it could support `json.Number` in addition though.